src/librustc_mir/interpret/place.rs

```diff
@@ -594,6 +594,13 @@ where
             StaticKind::Promoted(promoted, promoted_substs) => {
                 let substs = self.subst_from_frame_and_normalize_erasing_regions(promoted_substs);
                 let instance = ty::Instance::new(place_static.def_id, substs);
+
+                // even after getting substs from the frame, this instance may still be
+                // polymorphic because ConstProp will try to polymorphic promoted MIR
```

Comment:
```suggestion
                // polymorphic because `ConstProp` will try to polymorphic promoted MIR.
```

Reply:
```suggestion
                // polymorphic because ConstProp will try to promote polymorphic MIR
```

Reply:
@oli-obk integrated your nit in my nit. :)